core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCacheFactory.java

```diff
@@ -18,18 +18,17 @@
  */
 package org.apache.accumulo.fate.zookeeper;
 
-import java.util.HashMap;
-import java.util.Map;
-
 import org.apache.accumulo.core.singletons.SingletonManager;
 import org.apache.accumulo.core.singletons.SingletonService;
 
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
 /**
  * A factory for {@link ZooCache} instances.
  */
 public class ZooCacheFactory {
-  // TODO: make this better - LRU, soft references, ...
-  private static Map<String,ZooCache> instances = new HashMap<>();
+  private static Cache<String,ZooCache> instances = Caffeine.newBuilder().maximumSize(5).build();
```

Comment:
Why a max size of 5?  What determines that number?

Reply:
I don't know - I think the use-case for multiple instances in a VM is to support testing. Normally that number should be 1.  Does a caffeine cache pre-allocate entries up to max size on construction? If not, then a larger number could be appropriate?

Reply:
It does not (`initialCapacity(c)` would do that).